homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,249 @@
+"""Weather entity for Apple WeatherKit integration."""
+
+from typing import Any, cast
+
+from apple_weatherkit import DataSetType
+
+from homeassistant.components.weather import (
+    Forecast,
+    SingleCoordinatorWeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    UnitOfLength,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import WeatherKitDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator: WeatherKitDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    async_add_entities([WeatherKitWeather(coordinator)])
+
+
+condition_code_to_hass = {
+    "BlowingDust": "windy",
```

Comment:
Please import and use our weather conditions constants for the Home Assistant values:
https://github.com/home-assistant/core/blob/aff49cb67abc376d18a58347f13542bfbad81b6c/homeassistant/components/weather/__init__.py#L91-L105